extensions/as2/deployment/src/main/java/org/apache/camel/quarkus/component/as2/deployment/As2Processor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.as2.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.LogConsoleFormatBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import org.apache.camel.component.as2.AS2ClientManagerEndpointConfiguration;
+import org.apache.camel.component.as2.AS2ServerManagerEndpointConfiguration;
+import org.apache.camel.component.as2.api.util.AS2Utils;
+import org.apache.camel.quarkus.component.as2.As2Recorder;
+import org.apache.camel.quarkus.core.deployment.spi.UnbannedReflectiveBuildItem;
+import org.jboss.jandex.IndexView;
+
+class As2Processor {
+
+    private static final String FEATURE = "camel-as2";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    RuntimeInitializedClassBuildItem runtimeInitializedClasses() {
+        return new RuntimeInitializedClassBuildItem(AS2Utils.class.getCanonicalName());
+    }
+
+    @BuildStep
+    UnbannedReflectiveBuildItem whitelistConfigurationClasses() {
+        return new UnbannedReflectiveBuildItem(
```

Comment:
Thanks for the suggestion. I'll look into option of fixing problem in its root in camel.

Reply:
I should have also added that in Camel 3.5, you no longer have to register the various  `EndpointConfiguration` classes for reflection. So that part can be removed too.

Reply:
I've removed it. Thanks

Reply:
@jamesnetherton 
Locally it worked without whitelisting, but in build log for this PR, I ss:
`[error]: Build step org.apache.camel.quarkus.component.support.policy.deployment.PolicyProcessor#bannedReflectiveClasses threw an exception: java.lang.IllegalStateException: The following classes should either be whitelisted via an UnbannedReflectiveBuildItem or they should not be registered for reflection via ReflectiveClassBuildItem: [org.apache.camel.component.as2.AS2ClientManagerEndpointConfiguration, org.apache.camel.component.as2.AS2ServerManagerEndpointConfiguration]`

Reply:
I didn't probably understood your previous comment correctly. To be sure:

- I have to fix reason of failed reflection for camel classes
- I can remove registration for reflection for camel, right?

Reply:
Urgh! Sorry for the confusion. I forgot I have not committed the crucial part that removes the policy extension yet.

Let me do on the camel-master branch ASAP and your changes should work ok afterwards. I'll ping you when it's done.

Reply:
@JiriOndrusek if you rebase on the latest work from `camel-master`, it should work for you now.